src/code/sys_matrix.c

```diff
@@ -698,47 +702,70 @@ void Matrix_Transpose(MtxF* mf) {
     mf->yz = temp;
 }
 
-void func_800D1FD4(MtxF* mf) {
+/**
+ * Replaces the rotation in the matrix at the top of the stack, keeping scale and translation untouched.
+ *
+ * Considering only the 3x3 part (ignoring the last row and column which are left untouched), assuming
+ * \f[ \texttt{sCurrentMatrix} = R \cdot S \f]
+ *
+ * with \f$
+ *  S = \begin{bmatrix}
+ *      \alpha & 0 & 0 \\
+ *      0 & \beta & 0 \\
+ *      0 & 0 & \gamma
+ *      \end{bmatrix}
```

Comment:
Btw this is how this doc looks in doxygen

<details>

<summary>
Click me (careful dark mode users the screenshot is white)
</summary>

![](https://421.es/doyu/2afjbx)

</details>

I'm not a fan of how bad it looks in the source code though

Reply:
It might just be me and I didn't mention it for this PR, but I really don't think we should be using LaTeX or write things specific to doxygen in general.
I get the point of doxygen for something like a library, but I honestly don't see anyone using it when working with a codebase like this. So it's a cool bonus if it just uses comments and documentation we would be writing anyway, but I find it unfortunate to be writing things specifically for doxygen instead of, say, having it in a more basic format which is meant to be read from the code itself (eg. with `|`s for a matrix layout).

Also worth noting that any doxygen syntax like this with `\` and such isn't generally parsable by IDEs so it just gets cut-off. Here is how this one looks like in vscode for instance:
![image](https://user-images.githubusercontent.com/32455037/145021234-162acb7e-fcb0-46dc-8024-127f092286e2.png)
I did look for an extension or setting that supports it, but couldn't find anything

Reply:
fwiw I agree, I would rather have something readable in the codebase directly over needing to generate and use doxygen.
It may seem counter productive since we added doxygen to the project, but I see it as an optional thing to view the documentation if you want, not the only way